contracts/token_mock/src/lib.rs

```diff
@@ -0,0 +1,127 @@
+
+#[near_bindgen]
+impl Contract {
+    #[init]
+    pub fn new_default_meta(
+        owner_id: ValidAccountId,
+        ft_name: std::string::String,
+        total_supply: U128,
+    ) -> Self {
+        Self::new(
+            owner_id,
+            total_supply,
+            FungibleTokenMetadata {
+                spec: FT_METADATA_SPEC.to_string(),
+                name: ft_name,
+                symbol: "EXAMPLE".to_string(),
+                icon: Some(DATA_IMAGE_SVG_NEAR_ICON.to_string()),
+                reference: None,
+                reference_hash: None,
+                decimals: 24,
+            },
+        )
+    }
+
+    #[init]
+    fn new(owner_id: ValidAccountId, total_supply: U128, metadata: FungibleTokenMetadata) -> Self {
+        assert!(!env::state_exists(), "Already initialized");
+        metadata.assert_valid();
+        let mut this = Self {
+            token: FungibleToken::new(b"a".to_vec()),
+            metadata: LazyOption::new(b"m".to_vec(), Some(&metadata)),
+            owner_id: owner_id.clone(),
+        };
+        this.token.internal_register_account(owner_id.as_ref());
+        this.token
+            .internal_deposit(owner_id.as_ref(), total_supply.into());
+        this
+    }
+
+    pub fn give_tokens_to(&mut self, receiver_id: ValidAccountId, amount: U128) {
+        if !self.token.accounts.contains_key(receiver_id.as_ref()) {
+            self.token.internal_register_account(receiver_id.as_ref());
+        }
+
+        self.token.internal_transfer(
+            self.owner_id.as_ref(),
+            receiver_id.as_ref(),
+            amount.into(),
+            None,
+        );
+
+        log!("Gived {:?} tokens to account @{}", amount, receiver_id);
+        log!("Supply left {}", self.token.total_supply);
+        log!(
+            "Full balance {:?}",
+            self.token.accounts.get(receiver_id.as_ref())
+        );
+    }
+
+    // Duplicate of ERC20 interface (which is current NEAR FT)
+    pub fn internal_unwrap_balance_of(&mut self, account_id: &AccountId) -> Balance {
+        self.token.internal_unwrap_balance_of(account_id)
+    }
+
+    pub fn internal_deposit(&mut self, account_id: &AccountId, amount: Balance) {
+        self.token.internal_deposit(account_id, amount);
+    }
+
+    pub fn internal_withdraw(&mut self, account_id: &AccountId, amount: Balance) {
+        self.token.internal_withdraw(account_id, amount);
+    }
+
+    pub fn internal_transfer(
```

Comment:
@convowork1 interface should follow NEP21 standart https://nomicon.io/Standards/FungibleToken/Core.html

`internal_transfer -> ft_transfer`
`sender_id` should be implicit, this is id of the account that called `ft_transfer` could be retrived from the `env`